working/augmentation-libraries/feature-specification.md

```diff
@@ -722,11 +722,9 @@ It is a **compile-time error** if:
     const variable be augmented by another const variable, changing its value,
     or is that too weird?)**
 
-*   An `abstract` variable is augmented with a non-abstract variable, getter,
-    or setter.
+*   An `abstract` variable is augmented with a non-abstract variable.
 
-*   An `external` declaration is augmented with an `abstract` declaration. For
-    variables this also applies to the implicit getter and setter.
+*   An `external` variable is augmented with an `abstract` variable.
```

Comment:
Even that could be allowed and make sense:

```dart
external int x;

@metadata
augment abstract x;
```

Arguably, to avoid redundancy, we need to use an abstract variable because this allows us to associate the metadata with both the getter and the setter of the definition of `x`. Alternatively, we would do this:

```dart
external int x;

@metadata
augment get x;
@metadata
augment set x(_);
```

We could also repeat the `external` declaration, but this seems more confusing to me. "Does this mean that we have two native implementations of the getter and setter, and we're ignoring the first one?":

```dart
external int x;

@metadata
augment external x;
```

We could do that, but it seems cleaner to me if we use `abstract` which is the "smallest" possible declaration of a variable: No getter implementation, no setter implementation, no storage, no initializer, just a way to refer to the declaration with the given name `x`. Used in this manner, an `abstract` variable will work for the purpose of adding metadata (and the like) with a concrete variable, with a property which was originally expressed as a getter and a setter, with an external variable, the works.

Reply:
> @metadata
> augment abstract x;

I see, I think I probably would prefer that users just repeat `external` in this case? Otherwise it seems like you are changing the field from external to abstract, which are different.

Reply:
Right ... I tried to describe my mental model about these things here: https://github.com/dart-lang/language/issues/4017#issuecomment-2276228341. It would require certain rules to change (mostly: things would be allowed that are currently specified as errors), but I think it's nice to have a simple and consistent way to understand the semantics of any given stack of declarations for a given property.